src/Code/Viewer.tsx

```diff
@@ -0,0 +1,32 @@
+import * as React from 'react';
+import { Box, IBox } from '../Box';
+import { codeStyles } from './styles';
+import { astToReact } from './utils/astToReact';
+import { parseCode } from './utils/parseCode';
+
+export interface IViewer extends IBox {
+  value: string;
+  language?: string;
+  showLineNumbers?: boolean;
+  inline?: boolean;
```

Comment:
What does this solve / why is it an issue worth spending time on?

Reply:
What's the point of providing a default value for boolean in this particular case?
Certainly there are proper use cases, i.e. when you use strict equality comparison or you do need to pass boolean values somewhere else, however in this case we don't have any of these. :P

Reply:
> . An undefined boolean

Boolean cannot be undefined :trollface: 

Anyway, I can make it so that both of them are boolean if undefined is provided.

Reply:
*Counter-philosophy mode on*

My philosophy @XVincentX is this: a "boolean" is either _truthy_ or _falsey_. Whether it is actually the value `true` or `false` is of little importance, since all logical operators (like `&&` and `||`) and the mother-of-all-logical-operators the conditional `if` statement only care about the truthyness or falseyness of the value.

```js
if (undefined || "cat" && 55) {
  print("true!")
}
```

These logical operators aren't _coercing_ their arguments to `Boolean` data types and returning a `Boolean` - they DGAF about the type. `||` returns its first or second argument and `&&` returns its first argument, regardless of the type!

To me, this seems reasonable also because of JavaScript's C heritage. There is no boolean data type in C. In fact, there are no boolean data types in x86 assembly language. So to me, the `Boolean` type in JavaScript is a farce really.

Obviously... TypeScript has not been very understanding of my philosophy. :trollface: I think yesterday I literally made a type `boolean | null | undefined` (throw hands up in frustration).

*Looks at actual code being discussed*

Oh yeah, I totally agree with having fewer `?` types and more default values. Defaulting to `false` is a nice solution here.